Account/FirefoxAccount.swift

```diff
@@ -140,21 +149,26 @@ public class FirefoxAccount {
             configurationLabel = configurationLabel,
             email = dictionary["email"] as? String,
             uid = dictionary["uid"] as? String {
+                let fxaDeviceId = dictionary["fxaDeviceId"] as? String
+                let deviceRegistrationVersion = dictionary["deviceRegistrationVersion"] as? Int ?? 0
                 let stateCache = KeychainCache.fromBranch("account.state", withLabel: dictionary["stateKeyLabel"] as? String, withDefault: SeparatedState(), factory: stateFromJSON)
                 return FirefoxAccount(
                     configuration: configurationLabel.toConfiguration(),
                     email: email, uid: uid,
+                    fxaDeviceId: fxaDeviceId, deviceRegistrationVersion: deviceRegistrationVersion,
                     stateCache: stateCache)
         }
         return nil
     }
 
     public enum AccountError: MaybeErrorType {
         case NotMarried
+        case DeviceRegistrationFailed
```

Comment:
If we're handling device registration as a failure, does that mean for any successful device registration, we must have a deviceId/deviceRegistrationVersion? I wonder if we can strengthen the property types to be non-optional in this case.